merge.go

```diff
@@ -0,0 +1,528 @@
+		_, ok := mem.GetProperty(s)
+		if !ok {
+			delta.removeProp = append(delta.removeProp, s)
+		}
+		return true
+	})
+
+}
+
+// compares two lpg nodes and returns a diff
+func compareGraphNode(mem, db *lpg.Node, operationType bool) delta {
+	delta := delta{isNode: true, setProp: make(map[string]interface{})}
+	if db == nil {
+		delta.addLabel = mem.GetLabels().Slice()
+		delta.setProp = ls.CloneProperties(mem)
+		return delta
+	}
+	delta.addLabel = findNodeLabelDiff(mem.GetLabels(), db.GetLabels())
+	// Expected properties must be a subset
+	setPropertiesDelta(delta, mem, operationType)
+	if operationType == overwriteOp {
+		setNodeOverwrite(mem, db, delta)
+	}
+	return delta
+}
+
+// compares two native lpg nodes and returns a diff
+func compareGraphEdge(mem, db *lpg.Edge, operationType bool) delta {
+	delta := delta{isNode: false, setProp: make(map[string]interface{})}
+	if db == nil {
+		delta.addLabel = []string{mem.GetLabel()}
+		delta.setProp = ls.CloneProperties(mem)
+		return delta
+	}
+	if mem.GetLabel() != db.GetLabel() {
+		delta.addLabel = []string{mem.GetLabel()}
+	}
+	setPropertiesDelta(delta, mem, mergeOp)
+	if operationType == overwriteOp {
+		setEdgeOverwrite(mem, db, delta)
+	}
+	return delta
+}
+
+type updateNode struct {
+	id     int64
+	labels []string
+	props  map[string]interface{}
+	n      *lpg.Node
+	c      Config
+}
+
+type createNode struct {
+	id     int64
+	labels []string
+	props  map[string]interface{}
+	n      *lpg.Node
+	c      Config
+}
+
+type createEdge struct {
+	id               int64
+	typ              []string
+	props            map[string]interface{}
+	e                *lpg.Edge
+	fromNode, toNode int64
+	c                Config
+}
+
+type updateEdge struct {
+	id               int64
+	typ              []string
+	props            map[string]interface{}
+	e                *lpg.Edge
+	fromNode, toNode int64
+	c                Config
+}
+
+func buildQueriesFromDeltas(deltas []delta, cfg Config) []string {
+	res := make([]string, 0)
+	for _, d := range deltas {
+		switch d.operation {
+		case createNodeOp:
+			op := createNode{
+				n:      d.node,
+				id:     d.id,
+				labels: d.addLabel,
+				props:  d.setProp,
+				c:      cfg,
+			}
+			res = append(res, op.String())
+		case updateNodeOp:
+			op := updateNode{
+				n:      d.node,
+				id:     d.id,
+				labels: d.addLabel,
+				props:  d.setProp,
+				c:      cfg,
+			}
+			res = append(res, op.String())
+		case updateEdgeOp:
+			op := updateEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+				c:     cfg,
+			}
+			res = append(res, op.String())
+		case createEdgeOp:
+			op := createEdge{
+				e:     d.edge,
+				id:    d.id,
+				typ:   d.addLabel,
+				props: d.setProp,
+				c:     cfg,
+			}
+			res = append(res, op.String())
+		}
+	}
+	return res
+}
+
+func Merge(memGraph, dbGraph *lpg.Graph, dbGraphIds map[*lpg.Node]int64, dbEdges map[*lpg.Edge]int64, config Config) (*lpg.Graph, []string, error) {
```

Comment:
Merge should return []queryBuilders instead of []string

Have an option at the cobra cmd to print out the queries instead of executing them.